console/src/tasks.rs

```diff
@@ -249,14 +252,20 @@ impl State {
             return;
         }
 
-        self.tasks.retain(|_, task| {
-            let mut task = task.borrow_mut();
-            if task.completed_for == 0 {
-                return true;
-            }
-            task.completed_for += 1;
-            task.completed_for <= Self::RETAIN_COMPLETED_FOR
-        })
+        if let Some(now) = self.last_updated_at {
```

Comment:
Not sure whether this is a problem but this is not entirely accurate. The problem is that we calculate the `dopped_for` number based on the time of the last update.  So if no update is coming for a while we might see tasks sticking around longer than the configured `retain_for`

Reply:
I think that's fine, actually. We should expect updates to come pretty frequently (about once every second by default). In the previous implementation, we were only removing completed tasks after a particular number of updates had passed, so tasks would only be removed if we are continually receiving updates, and I think that's actually good --- if the connection temporarily goes down and we have to reconnect, I think it's nicer to keep showing whatever state we have until we are able to get back in touch with the app...